### PR TITLE
user route

### PR DESCRIPTION
**Pull request by jhchoi0830**

### CODE REVIEW

File: app/endpoints/car_routes.py [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 
 # import model
-from app.models.UsedCarModel import UsedCar
-from app.models.UserModel import User
+from app.models.usedCarModel import UsedCar
```
Comment (nhistory) [inline app/endpoints/car_routes.py:6]: Model naming convention
- UsedCarModel

File: app/endpoints/car_routes.py [hunk 2]
```diff
@@ -3,8 +3,8 @@
...
+from app.models.userModel import User
```
Comment (nhistory) [inline app/endpoints/car_routes.py:7]: Model naming conventions
- UserModel

**Review by nhistory — COMMENTED** [review]
There are no endpoints in user_routes.py

File: app/endpoints/car_routes.py (hunk 1 above)
Comment (aurora-yn) [reply app/endpoints/car_routes.py:6]: This is a file path. For the model, PascalCase is the correct convention. 
Not to break local server `usedCarModel` change is a right update. But, its file should be lower cases with underscores. I have made changes in my PR here: https://github.com/jhchoi0830/compcar_service/pull/12
After merging 12th PR, please update this branch `ENDPOINT-USERROUTE` before merge!

Let me know if you need help.

File: app/endpoints/user_routes.py [hunk 3]
```diff
@@ -2,7 +2,42 @@
...
+
+from app.services.database import (
+    fetch_all_users,
+    fetch_user_by_email,
+    fetch_user_by_password
+    )
+
+router = APIRouter()
+
+@router.get("/")
```
Comment (aurora-yn) [inline app/endpoints/user_routes.py]: Root endpoint already exist at `car_routes.py`. No needed in this route file.

File: app/endpoints/user_routes.py [hunk 4]
```diff
@@ -2,7 +2,42 @@
...
+async def root():
+    return {"message": "Hello World"}
+
+
+@router.get("/api/car")
```
Comment (aurora-yn) [inline app/endpoints/user_routes.py]: Please remove `/api/car` endpoint

File: app/endpoints/user_routes.py [hunk 5]
```diff
@@ -2,7 +2,42 @@
...
+async def get_user() -> list:
+    response = await fetch_all_users()
+    return response
+
+
+@router.get("/register/user/email/{email}", response_model=list[User])
```
Comment (aurora-yn) [inline app/endpoints/user_routes.py]: We will need POST method. Something like:
```
from app.services.database import create_user

@router.post('/api/user/register', response_model=User)
async def post_user(user: User):
    response = await create_user(user.dict())
    if response:
        return response
    raise HTTPException(400, "Something went wrong")

```
Also, please make sure if `user` collection is set in MongoDB.

File: app/services/database.py [hunk 6]
```diff
@@ -42,4 +43,28 @@ async def fetch_car_by_color(color: str) -> list:
     cursor = car_collection.find({"color":color})
     async for document in cursor:
         cars.append(document)
-    return cars
\ No newline at end of file
+    return cars
+
+
```
Comment (aurora-yn) [inline app/services/database.py]: We will need `create_user`.
Base structure will be:
```
async def create_user(user: User):
    result = await user_collection.insert_one(user)
    created_user = await user_collection.find_one({'_id': result.inserted_id})
    return created_user
```

**Review by aurora-yn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**